src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,168 @@
+                }
+                else if (reader.ValueTextEquals(NativePropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.NativeLibraries = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(BuildPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.Build = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(BuildMultiTargetingPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.BuildMultiTargeting = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(ContentFilesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.ContentFiles = reader.ReadObjectAsList<LockFileContentFile>(Utf8JsonReaderExtensions.LockFileContentFileConverter);
+                }
+                else if (reader.ValueTextEquals(RuntimeTargetsPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.RuntimeTargets = reader.ReadObjectAsList<LockFileRuntimeTarget>(Utf8JsonReaderExtensions.LockFileRuntimeTargetConverter);
+                }
+                else if (reader.ValueTextEquals(ToolsPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.ToolsAssemblies = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(EmbedPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.EmbedAssemblies = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(FrameworkReferencesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.FrameworkReferences = reader.ReadStringArrayAsIList();
+                }
+                else
+                {
+                    reader.Skip();
+                }
+            }
+            lockFileTargetLibrary.Freeze();
+            return lockFileTargetLibrary;
+        }
+
+        private IList<PackageDependency> ReadPackageDependencyList(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                return Array.Empty<PackageDependency>();
+            }
+
+            var packageDependencies = new List<PackageDependency>();
```

Comment:
@nkolev92 / @zivkan  do you know of a good way to determine what the most common value is for the number of packageDependencies? 

[Original Comment](https://github.com/NuGet/NuGet.Client/pull/5529#discussion_r1414975442) 
> Better default capacity based on most common value? This is defaulting to 4. 
-@davkean

Reply:
It's difficult to validate whatever you chose is a good value. I've attempted to choose one before and here's what I've considered:

- I added a logger in the code that printed the number of dependencies. I ran restore for a few different repos and looked at the results. Median varied between 2-4 for different repos, and average was between 3-5. 
- Look at the average number of dependencies for packages on nuget.org and got something like 3.1. 

See discussion: https://github.com/NuGet/NuGet.Client/pull/5531. 

I think the defaults are ok, but I'm a lot less knowledgeable about this than others :)

Reply:
3.1 is close enough to 4 that I feel comfortable leaving it as is.

Reply:
I looked at the trace, 6% of allocations is this collection being resized, so it feels like I would favor a little higher.

Reply:
Switched it to 10.

Reply:
> I looked at the trace, 6% of allocations is this collection being resized, so it feels like I would favor a little higher.

How do we decide how to trade off allocations unnecessarily vs resizing? Why is resizing worse than allocation extra unnecessarily?

Reply:
fwiw, the above dependencies count calculation does not account for how many dependencies a project itself references. That's probably higher, but unfortunately we don't have data for that.

So wouldn't shock me if higher than that works better.